apps/examples/wifi_manager_sample/wm_test_stress_test3.c

```diff
@@ -0,0 +1,280 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License\n");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include <stress_tool/st_perf.h>
+#include "wm_test.h"
+#include "wm_test_utils.h"
+#include "wm_test_network.h"
+#include "wm_test_log.h"
+
+#define WM_TEST_TRIAL 3
+#define WT_TAG "[WT]"
+
+static char *WM_AP_SSID;
+static char *WM_AP_PASSWORD;
+static wifi_manager_ap_auth_type_e WM_AP_AUTH;
+static wifi_manager_ap_crypto_type_e WM_AP_CRYPTO;
+static char *WM_SOFTAP_SSID;
+static char *WM_SOFTAP_PASSWORD;
+static int WM_SOFTAP_CHANNEL;
+static sem_t g_wm_sem;
+
+/*
+ * callbacks
+ */
+static void wm_sta_connected(wifi_manager_result_e);
+static void wm_sta_disconnected(wifi_manager_disconnect_e);
+static void wm_softap_sta_join(void);
+static void wm_softap_sta_leave(void);
+static void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_sta_connected,
+	wm_sta_disconnected,
+	wm_softap_sta_join,
+	wm_softap_sta_leave,
+	wm_scan_done,
+};
+
+#define WM_TEST_SIGNAL							\
+	do {										\
+		sem_post(&g_wm_sem);					\
+		WT_LOG(WT_TAG, "send signal");			\
+	} while (0)
+
+#define WM_TEST_WAIT							\
+	do {										\
+		WT_LOG(WT_TAG, "wait signal");			\
+		sem_wait(&g_wm_sem);					\
```

Comment:
Is it okay not to handle the result of `sem_wait`?
Sometimes, it can be failed to wait the semaphore.